### ISSUE TITLE
Create constants project to share between product and test code

### ISSUE THREAD

**Issue by derrickstolee** [Microsoft/VFSForGit#650]
We frequently have developer headaches when we update a string literal in the product code but forget to update it in the tests. This is particularly painful when the test only runs in the full functional test suite.

Create a new project full of constants that we can use across product and test code.

**Comment by kewillford** [comment]
I understand the frustration here but my big concern would be leaking product code into the test code so that it is not isolated.  Functional tests should not know anything about the internals of the product so that the tests will fail if something in the product changes.  Strings might not be a big deal but what about utility methods that are the same in both?  And soon more and more of your code is in the shared project and bugs get through because the tests aren't in isolation.  I think this is more something that we need to learn.  When we change a string, search for it in tests.

**Comment by derrickstolee** [comment]
I agree that we should be careful with what is shared, but this is not impossible to do safely.

I think this is the same idea as a compat layer between client and server. They need to be speaking the same language, even if they are formally separated.
